lib/rules/jsx-quotes.js

```diff
@@ -19,11 +21,19 @@ var astUtils = require("../ast-utils");
 var QUOTE_SETTINGS = {
     "prefer-double": {
         quote: "\"",
-        description: "singlequote"
+        alternateQuote: "'",
+        description: "singlequote",
+        convert: function(str) {
+            return toDoubleQuotes(str);
```

Comment:
oh, I got it the other way around, thanks.

Now, having said that, it looks like this rule can fails only in simple cases, that is: external quotes are wrong, and no other kind of quotes appear inside (escaped or not):
- `"..."` &rarr; prefer-single
- `'...'` &rarr; prefer-double

I can't find any counterexample to this. Am I missing something?
If that's the case, the fixes for this rule can simply be:

-  `str.replace(/"/g, "'")` (for prefer-single)
-  `str.replace(/'/g, "\"")` (for prefer-double)

That works because if any literal quote appears inside, it will make the rule pass even if the external quotes are not the preferred ones.